### PR TITLE
[IE TESTS] Conformance migration to OV2.0

### PR DESCRIPTION
**Pull request by iefode**
### Details:
 - *Conformance migration to OpenVINO 2.0 API (`SubgraphBaseClass` Inheritance)*
 - *Changing of default mechanism of input generation: To look up to the node type after parameter (As for `Conformance tests` in the previous version)*
 - *Changing of default mechanism of results comparation:  To look up to the node type before results (As for `Conformance tests` in the previous version)*
 - *Custom comparation was removed for NMS in default way (Removed from Conformance, not CPU SLT)*

### Tickets:
 - *69578*

### CODE REVIEW

File: src/tests/functional/plugin/cpu/single_layer_tests/non_max_suppression.cpp [hunk 1]
```diff
@@ -90,42 +90,42 @@ class NmsLayerCPUTest : public testing::WithParamInterface<NmsParams>, virtual p
...
-
-            inputs.insert({funcInput.get_node_shared_ptr(), tensor});
-        }
-    }
-    void compare(const std::vector<ov::runtime::Tensor> &expected, const std::vector<ov::runtime::Tensor> &actual) override {
-        CompareBBoxes(expected, actual);
-        inferRequestNum++;
-    }
-
+//    void generate_inputs(const std::vector<ngraph::Shape>& targetInputStaticShapes) override {
```
Comment (ilyachur) [inline src/tests/functional/plugin/cpu/single_layer_tests/non_max_suppression.cpp:93]: Please remove commented code

Comment (iefode) [reply src/tests/functional/plugin/cpu/single_layer_tests/non_max_suppression.cpp:93]: Marked as `Ready to review` just to check CI build and Conformance health. The PR is in progress yet:) I am going to remove all debug and commented code

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp [hunk 2]
```diff
@@ -0,0 +1,675 @@
...
+#include "shared_test_classes/single_layer/psroi_pooling.hpp"
+#include "shared_test_classes/base/utils/generate_inputs.hpp"
+#include "shared_test_classes/base/utils/ranges.hpp"
+
+namespace ov {
+namespace test {
+namespace utils {
+
+namespace {
+ov::runtime::Tensor generate(const std::shared_ptr<ov::Node> node,
```
Comment (ilyachur) [inline src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:19]:
```suggestion
ov::runtime::Tensor generate(const std::shared_ptr<ov::Node>& node,
```

File: src/tests/functional/shared_test_classes/include/shared_test_classes/base/utils/compare_results.hpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+#include <ie_core.hpp>
+#include "ngraph/node.hpp"
+
+namespace ov {
+namespace test {
+namespace utils {
+
+using CompareMap = std::map<ov::NodeTypeInfo, std::function<void(
+        const std::shared_ptr<ov::Node> node,
```
Comment (ilyachur) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/base/utils/compare_results.hpp:15]:
```suggestion
        const std::shared_ptr<ov::Node>& node,
```

File: src/tests/functional/shared_test_classes/include/shared_test_classes/base/utils/generate_inputs.hpp [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+#include <ie_core.hpp>
+#include "ngraph/node.hpp"
+
+namespace ov {
+namespace test {
+namespace utils {
+
+using InputsMap = std::map<ov::NodeTypeInfo, std::function<ov::runtime::Tensor(
+        const std::shared_ptr<ov::Node> node,
```
Comment (ilyachur) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/base/utils/generate_inputs.hpp:15]:
```suggestion
        const std::shared_ptr<ov::Node>& node,
```

File: src/tests/functional/shared_test_classes/src/base/utils/compareResults.cpp [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+#include "shared_test_classes/base/layer_test_utils.hpp"
+#include "shared_test_classes/base/utils/compare_results.hpp"
+#include "functional_test_utils/ov_tensor_utils.hpp"
+
+namespace ov {
+namespace test {
+namespace utils {
+
+namespace {
+void compare(const std::shared_ptr<ov::Node> node,
```
Comment (ilyachur) [inline src/tests/functional/shared_test_classes/src/base/utils/compareResults.cpp:16]:
```suggestion
void compare(const std::shared_ptr<ov::Node>& node,
```

File: src/tests/ie_test_utils/common_test_utils/data_utils.hpp [hunk 6]
```diff
@@ -112,32 +105,45 @@ fill_data_roi(InferenceEngine::Blob::Ptr &blob, const uint32_t range, const int
...
-        data[i + 3] = static_cast<dataType>(is_roi_max_mode ? std::floor(x1) : x1);
-        if (data[i + 3] < data[i + 1]) {
-            std::swap(data[i + 1], data[i + 3]);
-        }
-        if (data[i + 1] < 0)
-            data[i + 1] = 0;
-        if (data[i + 3] > max_x)
-            data[i + 3] = static_cast<dataType>(max_x);
+    for (size_t i = 0; i < data_size; i += 5) {
+    data[i] = static_cast<dataType>(distribution(random));
```
Comment (ilyachur) [inline src/tests/ie_test_utils/common_test_utils/data_utils.hpp:109]: Incorrect code style

**Comment by iefode** [comment]
@ilyachur Thank a lot for review :)

File: src/tests/ie_test_utils/common_test_utils/data_utils.hpp (hunk 6 above)
Comment (iefode) [reply src/tests/ie_test_utils/common_test_utils/data_utils.hpp]: What exactly do you mean? `dataType`?)

Comment (ilyachur) [reply src/tests/ie_test_utils/common_test_utils/data_utils.hpp]: I mean indentation :)

**Comment by iefode** [comment]
@ilyachur Could you please review again, if it is possible :)

**Review by andrei-kochin — APPROVED** [review]

File: src/tests/functional/plugin/conformance/test_runner/conformance_infra/include/conformance.hpp [hunk 7]
```diff
@@ -6,15 +6,17 @@
...
 #include <common_test_utils/file_utils.hpp>
 
-namespace ConformanceTests {
+namespace ov {
+namespace test {
+namespace conformance {
 
-extern const char* targetDevice;
-extern const char* targetPluginName;
+extern const char *targetDevice;
```
Comment (ilyachur) [inline src/tests/functional/plugin/conformance/test_runner/conformance_infra/include/conformance.hpp:13]: Why C style? Why not std::string?

Comment (iefode) [reply src/tests/functional/plugin/conformance/test_runner/conformance_infra/include/conformance.hpp:13]: that was rule of code style :(

File: src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp [hunk 8]
```diff
@@ -130,8 +131,22 @@ void SubgraphBaseTest::query_model() {
...
-    for (size_t i = 0; i < expected.size(); i++) {
-        ov::test::utils::compare(expected[i], actual[i], abs_threshold, rel_threshold);
+    ASSERT_EQ(expected.size(), function->get_results().size());
+    auto compareMap = utils::getCompareMap();
+    const auto& results = function->get_results();
+    for (size_t j = 0; j < results.size(); j++) {
+        const auto result = results[j];
+        for (size_t i = 0; i < result->get_input_size(); ++i) {
+            std::shared_ptr<ov::Node> inputNode = result->get_input_node_shared_ptr(i);
+            if (ngraph::is_type<ov::op::v0::Convert>(inputNode)) {
```
[the remark this answers is not in the document]
Comment (iefode) [reply src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:141]: do you mean `type_info` or `dynamic_cast`?

File: src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp [hunk 9]
```diff
@@ -4,9 +4,10 @@
 
 #include <signal.h>
 #include <fstream>
-#include <transformations/utils/utils.hpp>
-#include <transformations/convert_precision.hpp>
+#include <shared_test_classes/base/utils/generate_inputs.hpp>
```
[the remark this answers is not in the document]
Comment (iefode) [reply src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp]: Fixed, thanks)

**Pull request merged**